add hmac signature

pheenomenon opened the pull request:
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes # (issue)

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

sthyagarajan reviewed (approved):
LGTM apart from a few nitpicks
1. `add a test when private key is not there and establish the expected behavior`
2.  verify on hmac-signature.js check if it needs to be moved to test libs but its fine where it is

pheenomenon commented:
> LGTM apart from a few nitpicks
> 
> 1. `add a test when private key is not there and establish the expected behavior`
> 2. verify on hmac-signature.js check if it needs to be moved to test libs but its fine where it is

Thanks, to be consistent, changed logic to throw error if private key doesn't exist. Added test for it.
In two minds about `verify` in hmac-signature.js, its currently only used in test, but will leave it as is for readability and completion.
